packages/blade/src/tokens/theme/theme.d.ts

```diff
@@ -155,6 +155,7 @@ export type Colors = {
 export type ColorsWithModes = Record<ColorSchemeModes, Colors>;
 
 export type ThemeTokens = {
+  name: 'paymentTheme' | 'bankingTheme' | (string & Record<never, never>); // Can be used to watch over state changes between theme without watching over entire theme object
```

Comment:
Interesting, what does `| (string & Record<never, never>)` do?

Reply:
This makes the union accept "any" string but also shows intellisense. 

Say you have `createTheme()` function which takes a name arg as `string` you want it to be "paymentTheme | bankingTheme | string"

```ts
type Name = "paymentTheme" | "bankingTheme" | string;

function createTheme(name: Name) {}

createTheme("bankingTheme")
createTheme("or-any-random-name");
```

but now users won't get the autocomplete for the `bankingTheme | paymentTheme` in the createTheme(), this happens because `string` is a super type of both  literal string `bankingTheme` & `paymentTheme` so typescript compiler simplifies the type to just `string` 

<img width="726" alt="image" src="https://user-images.githubusercontent.com/35374649/222077242-74bccef7-67e4-4957-8dbe-38a8f1f7540c.png">


But `adding (string & {})` makes it so that the union can accept any string but also tricks the typescript compiler to show the intellisense 

<img width="743" alt="image" src="https://user-images.githubusercontent.com/35374649/222077450-7d1ef6ca-13ca-4578-a612-f11432a495de.png">

[TSPlayground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAcghgW2gXigIjHESB2wAqAFhEmlAD7oBGcOA1gJY4DmRJEZlAFAM7ABOTZlABkUAN4BfAJQBuAFDyAZgFccAY2AMA9jijr+EOMAhskXHIggAuWFekTJi+QHoXUEBBABCKHBXA2uraCGAANhAm8gZGJmYQXGhocq7uUOkZGQB6APxQmNgQePEUUDT0QvGKMcamxOZoVAEAtLQgzfy0ACYhzZakKUA)


`(string & {})` works because when you intersect a the type `string` with the empty object `{}` typescript won't be able to simplify the the type to just `string`

Reply:
TIL 🤯

Reply:
So cool!

Reply:
Right now since we only support these two themes, should we remove accepting any other string value?

Reply:
I kept it for `overrideTheme` So if someone is overriding the theme, they can name it something different.